sql/src/main/java/io/crate/analyze/where/NullEliminator.java

```diff
@@ -57,13 +57,19 @@
 
     private static final Visitor VISITOR = new Visitor();
 
-    public static Symbol eliminateNullsIfPossible(Symbol symbol) {
-        return VISITOR.process(symbol, new Context());
+    public static Symbol eliminateNullsIfPossible(Symbol symbol,
+                                                  java.util.function.Function<Symbol, Symbol> postProcessor) {
```

Comment:
Maybe add a doc explaining that `postProcessor` is only applied on function nodes that changed due to null replacement.